### PR TITLE
[JBWS-4335] return picketbox support and maintain elytron support. Ch…

### PR DESCRIPTION
**Pull request by rsearls**
…ange CI wfly version

### CODE REVIEW

File: .github/workflows/maven.yml [hunk 1]
```diff
@@ -18,7 +18,7 @@ jobs:
       matrix:
         os: [ubuntu-latest, windows-latest ]
         java: ['8', '11']
-        wildfly-version: ['wildfly2400', 'wildfly2500']
+        wildfly-version: ['wildfly2300']
```
Comment (jimma) [inline .github/workflows/maven.yml:21]: I think the wildfly2400 could be much better and wildfly2500 can be kept. We have a excluded test list in https://issues.redhat.com/browse/JBWS-4230

File: modules/server/src/main/java/org/jboss/wsf/stack/cxf/security/authentication/SubjectCreatingInterceptor.java [hunk 2]
```diff
@@ -59,21 +60,6 @@
...
-        try
-        {
-            clazz = Class.forName("java.security.acl.Group");
-        } catch (Throwable t)
-        {
-            // ignore
-        }
-        groupClass = clazz;
-    }
-
```
Comment (jimma) [inline modules/server/src/main/java/org/jboss/wsf/stack/cxf/security/authentication/SubjectCreatingInterceptor.java:76]: Why are these lines removed ? This is for https://issues.redhat.com/browse/JBWS-4133.

File: modules/server/src/main/java/org/jboss/wsf/stack/cxf/security/authentication/SubjectCreatingInterceptor.java [hunk 3]
```diff
@@ -185,7 +171,7 @@ protected void setSubject(String name, String password, boolean isDigest, String
    private boolean checkUserPrincipal(Set<Principal> principals, String name)
    {
       for (Principal p : principals) {
-         if (groupClass == null || !groupClass.isInstance(p)) {
+         if (!(p instanceof Group)) {
```
Comment (jimma) [inline modules/server/src/main/java/org/jboss/wsf/stack/cxf/security/authentication/SubjectCreatingInterceptor.java:174]: Same for https://issues.redhat.com/browse/JBWS-4133.

File: modules/server/src/main/java/org/jboss/wsf/stack/cxf/security/authentication/SubjectCreatingPolicyInterceptor.java [hunk 4]
```diff
@@ -163,7 +149,7 @@ protected Principal getPrincipal(Principal originalPrincipal, Subject subject)
       if (!principals.isEmpty())
       {
           Principal principal = principals.iterator().next();
-          if (groupClass == null || !groupClass.isInstance(principal))
+          if (!(principal instanceof Group))
```
Comment (jimma) [inline modules/server/src/main/java/org/jboss/wsf/stack/cxf/security/authentication/SubjectCreatingPolicyInterceptor.java:152]: Same for https://issues.redhat.com/browse/JBWS-4133.

File: modules/testsuite/cxf-tests/src/test/scripts/jbws-testsuite-default-elytron.groovy [hunk 5]
```diff
@@ -11,6 +11,120 @@ if (!session.userProperties['enableServerLoggingToConsole'] && !project.properti
...
+loginModuleDigest.appendNode('module-option', ['name':'storeDigestCallback','value':'org.jboss.wsf.stack.cxf.security.authentication.callback.UsernameTokenCallback'])
+loginModuleDigest.appendNode('module-option', ['name':'hashStorePassword','value':'true'])
+loginModuleDigest.appendNode('module-option', ['name':'unauthenticatedIdentity','value':'anonymous'])
+loginModuleDigest.appendNode('module-option', ['name':'usersProperties','value':testResourcesDir + '/jaxws/samples/wsse/policy/jaas/digest/WEB-INF/jbossws-users.properties'])
+loginModuleDigest.appendNode('module-option', ['name':'rolesProperties','value':testResourcesDir + '/jaxws/samples/wsse/policy/jaas/digest/WEB-INF/jbossws-roles.properties'])
+
+
+
+
+
```
Comment (jimma) [inline modules/testsuite/cxf-tests/src/test/scripts/jbws-testsuite-default-elytron.groovy:127]: We'd better add this in a groovy script dedicated to configure legacy subsystem to run the tests instead of get all these things mixed in both legacy and elytron configuration.

File: modules/testsuite/cxf-tests/src/test/java/org/jboss/test/ws/jaxws/samples/wsse/policy/jaspi/UsernamePasswordCallback.java [hunk 6]
```diff
@@ -35,7 +35,7 @@ public void handle(Callback[] callbacks) throws IOException, UnsupportedCallback
    {
       WSPasswordCallback pc = (WSPasswordCallback)callbacks[0];
       if ("kermit".equals(pc.getIdentifier()))
-         pc.setPassword("thefrog");
+         pc.setPassword("therealfrog");
```
Comment (jimma) [inline modules/testsuite/cxf-tests/src/test/java/org/jboss/test/ws/jaxws/samples/wsse/policy/jaspi/UsernamePasswordCallback.java:38]: Did all things https://github.com/jbossws/jbossws-cxf/commit/f275c43931b9113d393a2c02e82ace27e1d9029c back ported ?

File: .github/workflows/maven.yml (hunk 1 above)
Comment (rsearls) [reply .github/workflows/maven.yml]: This is not going to run on tag wildfly-2400 either.  The only wildfly-24 branch that has the elytron changes is origin/24.x

File: modules/server/src/main/java/org/jboss/wsf/stack/cxf/security/authentication/SubjectCreatingInterceptor.java (hunk 2 above)
Comment (rsearls) [reply modules/server/src/main/java/org/jboss/wsf/stack/cxf/security/authentication/SubjectCreatingInterceptor.java:76]: changed

File: modules/server/src/main/java/org/jboss/wsf/stack/cxf/security/authentication/SubjectCreatingInterceptor.java (hunk 3 above)
Comment (rsearls) [reply modules/server/src/main/java/org/jboss/wsf/stack/cxf/security/authentication/SubjectCreatingInterceptor.java]: changed

File: modules/server/src/main/java/org/jboss/wsf/stack/cxf/security/authentication/SubjectCreatingPolicyInterceptor.java (hunk 4 above)
Comment (rsearls) [reply modules/server/src/main/java/org/jboss/wsf/stack/cxf/security/authentication/SubjectCreatingPolicyInterceptor.java]: changed

File: modules/testsuite/cxf-tests/src/test/scripts/jbws-testsuite-default-elytron.groovy (hunk 5 above)
Comment (rsearls) [reply modules/testsuite/cxf-tests/src/test/scripts/jbws-testsuite-default-elytron.groovy]: changed

File: modules/testsuite/cxf-tests/src/test/java/org/jboss/test/ws/jaxws/samples/wsse/policy/jaspi/UsernamePasswordCallback.java (hunk 6 above)
Comment (rsearls) [reply modules/testsuite/cxf-tests/src/test/java/org/jboss/test/ws/jaxws/samples/wsse/policy/jaspi/UsernamePasswordCallback.java:38]: the value is in the property file was changed to "therealfrog" sometime in the past.  This is the correct value.

File: .github/workflows/maven.yml (hunk 1 above)
Comment (jimma) [reply .github/workflows/maven.yml]: what do you mean by wildfly 2400 tag ? Is it WFLY 24.0.0.Final ?

File: modules/testsuite/cxf-tests/src/test/java/org/jboss/test/ws/jaxws/samples/wsse/policy/jaspi/UsernamePasswordCallback.java (hunk 6 above)
Comment (jimma) [reply modules/testsuite/cxf-tests/src/test/java/org/jboss/test/ws/jaxws/samples/wsse/policy/jaspi/UsernamePasswordCallback.java:38]: This is the part of change in  https://github.com/jbossws/jbossws-cxf/commit/f275c43931b9113d393a2c02e82ace27e1d9029c.  Did the change in `SubjectCreator.java` backport to this PR ?

File: .github/workflows/maven.yml (hunk 1 above)
Comment (rsearls) [reply .github/workflows/maven.yml]: In the wildfly project there are 3 tagged wfly-24 versions, 24.0.0.Beta1, 
24.0.0.Final, 24.0.1.Final.  None of these tagged versions contains the
elyron changes in module webservices or the version update to jbossws-spi.
Currently branch 24.x is the only one that contains these changes.  Its version
number is 24.0.2.Final-SNAPSHOT.

I have been unsuccessful changing this yml file to checkout the 24.x branch and
build the project so that jbossws-cxf can use it to run the tests.

File: modules/testsuite/cxf-tests/src/test/java/org/jboss/test/ws/jaxws/samples/wsse/policy/jaspi/UsernamePasswordCallback.java (hunk 6 above)
Comment (rsearls) [reply modules/testsuite/cxf-tests/src/test/java/org/jboss/test/ws/jaxws/samples/wsse/policy/jaspi/UsernamePasswordCallback.java:38]: Yes the code change from f275c43 is present in SubjectCreator.

File: modules/testsuite/cxf-tests/pom.xml [hunk 7]
```diff
@@ -498,6 +498,272 @@
...
+                          </execution>
+                          <execution>
+                              <id>testsuite-ssl-mutual-auth</id>
+                              <phase>pre-integration-test</phase>
+                              <goals>
+                                  <goal>execute</goal>
+                              </goals>
+                              <configuration>
+                                  <scripts>
+                                      <script>file:///${basedir}/src/test/scripts/jbws-testsuite-ssl-mutual-auth-elytron.groovy</script>
```
[the remark this answers is not in the document]
Comment (rsearls) [reply modules/testsuite/cxf-tests/pom.xml:559]: There were no changes needed to the server configuration related to picketbox
for this category of tests.  I chose to maintain the reference to the original 
file and not make a duplicate with just a name change.

The only reason I maintained the "-elytron" designator from the (original) files 
was to relate back your original naming scheme.  I felt this would provide clarity 
to the user community that was familiar with the old naming convention.

File: modules/testsuite/cxf-tests/pom.xml [hunk 8]
```diff
@@ -498,6 +498,272 @@
...
+                          </execution>
+                          <execution>
+                              <id>testsuite-default-config-tests</id>
+                              <phase>pre-integration-test</phase>
+                              <goals>
+                                  <goal>execute</goal>
+                              </goals>
+                              <configuration>
+                                  <scripts>
+                                      <script>file:///${basedir}/src/test/scripts/jbws-testsuite-default-config-tests-elytron.groovy</script>
```
[the remark this answers is not in the document]
Comment (rsearls) [reply modules/testsuite/cxf-tests/pom.xml:593]: There were no changes needed to the server configuration related to picketbox
for this category of tests.  I chose to maintain the reference to the original 
file and not make a duplicate with just a name change.

File: modules/testsuite/cxf-tests/pom.xml (hunk 7 above)
[the remark this answers is not in the document]
Comment (jimma) [reply modules/testsuite/cxf-tests/pom.xml:559]: @rsearls  But this groovy script is to create the tls element under elytron subsystem for https, and it all uses the elytron configuration instead of picketbox.

File: modules/testsuite/cxf-tests/pom.xml (hunk 8 above)
[the remark this answers is not in the document]
Comment (jimma) [reply modules/testsuite/cxf-tests/pom.xml:593]: @rsearls  This script is to create elytron configuration and enable the elytron in undertow and ejb to test.  We are going to test with picketbox, we have to configure it for test and it can't coexist in undertow and ejb subsystems.
